uportal-war/src/main/java/org/jasig/portal/utils/LdapQueryUtils.java

```diff
@@ -0,0 +1,34 @@
+package org.jasig.portal.utils;
+
+public class LdapQueryUtils {
+
+	/**
+	 * 
+	 * @param query
+	 * @return
+	 */
+	public static String escapeSpecialCharacters(String query) {
+        // We need to escape regex special characters that appear in the query string...
+        final String[][] specials = new String[][] {
+                            /* backslash must come first! */
+                            new String[] { "\\", "\\\\"}, 
+                            new String[] { "[", "\\[" }, 
+                            /* closing ']' isn't needed b/c it's a normal character w/o a preceding '[' */
+                            new String[] { "{", "\\{" }, 
+                            /* closing '}' isn't needed b/c it's a normal character w/o a preceding '{' */
+                            new String[] { "^", "\\^" },
+                            new String[] { "$", "\\$" },
+                            new String[] { ".", "\\." },
+                            new String[] { "|", "\\|" },
+                            new String[] { "?", "\\?" },
+                            new String[] { "*", "\\*" },
+                            new String[] { "+", "\\+" },
+                            new String[] { "(", "\\(" },
+                            new String[] { ")", "\\)" }
+                        };
```

Comment:
In looking at the spec more closely and looking at some other examples, such as section 'Specifying Non 7-Bit ASCII Characters in Search Filters' at https://docs.oracle.com/cd/E19424-01/820-4811/gdxpo/index.html where it states:
> Non 7-bit ASCII characters in search filters must be replaced with a representation of the character, where each byte > of the UTF-8 encoding is preceded by a backslash. In UTF-8, characters are represented by a hexadecimal code for > each byte.

> For example, the character é has UTF-8 representation c3a9. Thus, in a search filter, you represent é as \\c3\\a9. So, > to search for cn=Véronique Martin:

> ldapsearch -h myServer -b "dc=example,dc=com" "(cn=V\\c3\\a9ronique Martin)"

You will also need to look at each character and if it is above 0x7F represent it as \\XX (for 0x00XX) or \\XX\\YY for 0xXXYY.  This is required for internationalization.